cliquet/tests/test_listeners.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+import mock
+from pyramid import testing
+import uuid
+import redis
+import json
+from datetime import datetime
+from contextlib import contextmanager
+
+from cliquet import initialization
+from cliquet.events import CollectionChanged
+from .support import unittest
+
+
+class ListenerSetupTest(unittest.TestCase):
+    def test_redis_listener_is_enabled_via_setting(self):
+        listener = 'cliquet.listeners.redis'
+        redis_class = mock.patch(listener + '.RedisListener')
+        config = testing.setUp()
+        with mock.patch.dict(config.registry.settings,
+                             [('event_listeners', listener)]):
+            with redis_class as redis_mocked:
+                initialization.setup_listeners(config)
+                self.assertTrue(redis_mocked.called)
+
+
+@contextmanager
+def broken_redis():
+    from redis import StrictRedis
+    old = StrictRedis.lpush
+
+    def push(*args, **kwargs):
+        raise Exception('boom')
+
+    StrictRedis.lpush = push
+    yield
+    StrictRedis.lpush = old
+
+
+class ListenerCalledTest(unittest.TestCase):
+
+    def setUp(self):
+        self.config = testing.setUp()
+        pool = redis.BlockingConnectionPool(max_connections=1)
+        self._redis = redis.StrictRedis(host='localhost', port=6739, db=0,
+                                        connection_pool=pool)
```

Comment:
Which load_from_config ? This is not to load a RedisListener. This is a redis client to verify values in Redis.